app/services/whatsapp/providers/whatsapp_cloud_service.rb

```diff
@@ -40,8 +44,21 @@ def media_url(media_id)
     "#{api_base_path}/v13.0/#{media_id}"
   end
 
+  def fetch_next_templates(response)
+    loop do
+      break unless response['paging'] && response['paging']['next']
+
+      response = HTTParty.get(response['paging']['next'])
+      save_templates(response)
+    end
+  end
+
   private
 
+  def save_templates(response)
+    whatsapp_channel.update(message_templates: response['data'], message_templates_last_updated: Time.now.utc) if response.success?
```

Comment:
Wouldn't this overwrite the previous template? I believe we should do a concat of all the responses here.

Reply:
Yeah figured that updating the PR. 👍🏼